src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -289,6 +304,23 @@ public final void apply(Project project) {
         });
     }
 
+    private static void guardConfigurationFromEarlyResolution(
+            Project project, NamedDomainObjectProvider<Configuration> configuration) {
+        ListProperty<DependencyConstraint> constraintsProperty =
+                GradleWorkarounds.fixListProperty(project.getObjects().listProperty(DependencyConstraint.class));
+        // Guard against bad plugins.
+        constraintsProperty.addAll(project.provider(() -> {
+            if (GradleWorkarounds.isConfiguring(project.getRootProject().getState())) {
+                throw new RuntimeException(String.format(
+                        "A plugin has realized the dependencies of %s while the project is still configuring. "
+                                + "This is not allowed when using GCV.",
+                        configuration.get()));
+            }
+            return ImmutableList.of();
```

Comment:
We need to return this so the resolution constraint gets triggered in `addAllLater`?

Reply:
yep, I didn't know how else to add this as an "effect" essentially